api/crud.py

```diff
@@ -193,23 +197,27 @@ async def invoice_add_related(item: models.Invoice):
     item.products = [product_id for product_id, in result if product_id]
     item.payments = []
     payment_methods = await models.PaymentMethod.query.where(models.PaymentMethod.invoice_id == item.id).gino.all()
+    currency_payment_methods = defaultdict(list)
     for method in payment_methods:
-        # TODO: multiple wallet same currency case
-        # TODO remove duplication
```

Comment:
By remove duplication I meant, see the `_create_payment_method` method, it contains the same code but misses name attribute.
Please fix this, maybe we can make a class method `format` or `json` (probably the latter) that will return json representation of a model. In general I think in the future we will need to abstract this more